engine/src/state_chain_observer/client.rs

```diff
@@ -288,9 +288,14 @@ impl<BaseRpcClient: base_rpc_api::BaseRpcApi + Send + Sync + 'static, SignedExtr
 
 			match block_compatibility.compatibility {
 				CfeCompatibility::NoLongerCompatible =>
-					return Err(
-						CreateStateChainClientError::CompatibilityError(block_compatibility).into()
-					),
+					if error_on_incompatible_block {
```

Comment:
This case should still error, or should faill thorugh the to wait_for_compatibly case otherwise we will return a incpmpatible block from the stream i think.

Reply:
Ah yep you're right, we do that check in between, now it'd panic, thought it'd fall through to be caught on the next part 👍 will fix

Reply:
Ok, its kind of hard, in this case (in the unfinalized stream case), if we get here, what we want to wait for the finalized stream to fail. We may need to remove the cache the from stream, so the unfinalized stream doesn't need an initial block.

Reply:
So that we don't need to check compatibility to create the unfinalized stream in the first place.

Reply:
We'd just filter the incompatible blocks. As we already do, below.